### ISSUE TITLE
Invalid state transition from Published to Published on publish refresh request

### ISSUE THREAD

**Issue by erogleva** [onsip/SIP.js#903]
**Describe the bug**
I am trying to send a PUBLISH request for the presence event using the Publisher class. The publish request is successful and I get a SIP 200 response. Shortly before the expiration time, a publish refresh request is sent, to which I also get a successful response; however there is an error about an invalid state transition. 

**To Reproduce (if possible)**
Send a PUBLISH request, then wait for the refresh request before its expiration.

**Expected behavior**
No errors or warnings on refresh.

**Observed behavior**
```
    index.js:1384 Error: Invalid state transition from Published to Published
    at invalidTransition (publisher.js:346)
    at Publisher.stateTransition (publisher.js:360)
    at Publisher.receiveResponse (publisher.js:196)
    at Object.onAccept (publisher.js:276)
    at PublishUserAgentClient.receiveResponse (user-agent-client.js:310)
    at Object.receiveResponse (user-agent-client.js:360)
    at NonInviteClientTransaction.receiveResponse (non-invite-client-transaction.js:105)
    at UserAgentCore.receiveResponseFromTransport (user-agent-core.js:883)
    at UserAgentCore.receiveIncomingResponseFromTransport (user-agent-core.js:206)
    at UserAgent.onTransportMessage (user-agent.js:974)
    at Transport.transport.onMessage (user-agent.js:840)
    at Transport.onWebSocketMessage (transport.js:491)
    at WebSocket.<anonymous> (transport.js:247)
```

**Environment Information**
 - sip.js 0.19.0
 - Chrome Version 88.0.4324.146

**Comment by slavikbialik** [comment]
Actually I am having the issue as you do, and just wanted to open a similar bug a few days ago. :)
So also joining to your question.
Experienced it on 0.18 and 0.19.

**Comment by erogleva** [comment]
Or maybe a state transition shouldn't be triggered at all when the response is received and the state is already Published?

https://github.com/onsip/SIP.js/blob/f11dfd584bc9788ccfc94e03034020672b738975/src/api/publisher.ts#L232

It seems that a similar issue is solved in this way in the registerer 
https://github.com/onsip/SIP.js/blob/b8c74916a0d6320b1137ede0c8d0afee594be9b3/src/api/registerer.ts#L688

**Comment by james-criscuolo** [comment]
Fixed this with the suggested solution here, thanks: https://github.com/onsip/SIP.js/commit/4627789d84844af3427b51768bf4bb390b855e8b

This code is not adequately tested, otherwise this would've been caught. Will close when this is released.

**Comment by erogleva** [comment]
Thanks!

**Comment by james-criscuolo** [comment]
Released in 0.20.0.
